Automattic/wp-calypso
# Purchases: clean up stored payment method components

#### Changes proposed in this Pull Request

What started as fixing an alignment issue on the Payment Methods screen turned into a full refactor :D. In this PR, I made the following updates:

- Updated the design of the payment method screen
- Consolidated all payment method components into a single folder (since the components are no longer shared).
- Remove credit card list component from devdocs because it is no longer a shared component.
- Removed the credit card component from the old checkout (which isn't used anymore).
- Cleanup a lot of redundant code.
- Renamed payment method components to be more generically applied to payment methods and also to be clearer as to what they do. 
- Updated out of date credit card icons

**Before**
![image](https://user-images.githubusercontent.com/6981253/98973961-5f005c00-24e2-11eb-995f-bfc274d0e507.png)

![image](https://user-images.githubusercontent.com/6981253/98973995-66c00080-24e2-11eb-860d-d788f11fa9a5.png)

**After**
![image](https://user-images.githubusercontent.com/6981253/98973888-4d1eb900-24e2-11eb-8c9a-bef3b1513c05.png)

![image](https://user-images.githubusercontent.com/6981253/98973916-5576f400-24e2-11eb-86d8-0803d7b73018.png)


#### Testing instructions

* Visit the payment method screens at the site (`/purchases/payment-methods/`) and account level (`/me/purchases/payment-methods`)
* Add a payment method and make sure it shows up.
* Make a purchase with a new payment method and confirm that it shows up correctly. 
* Take a peak on desktop + mobile to make sure nothing is broken.
